### ISSUE TITLE
ELI5: Why can't we have --lowvram

### ISSUE THREAD

**Issue by muunkky** [oobabooga/text-generation-webui#3314]
The goal of oobabooga is to be the a1111 of text based llms. I think a key factor in a1111's success was in the ability to run on low vram computers. I know there are hacks here an there to try to get it to work, but what was the method that makes --lowvram work on a1111 and why can't we have it on oobabooga?

**Comment by Ph0rk0z** [comment]
For one, image model vram usage is determined by the output size. While context will play a factor here, you can't really make an LLM take up less ram unless you use a smaller model or quantize to lower bits. Besides SD-XL, idle usage of SD is TINY.

**Comment by ddh0** [comment]
Are you using the latest release? I have a `low-vram` checkbox on the Model tab, and a similar CLI option. For llama.cpp backend.

**Comment by Ph0rk0z** [comment]
It's apples vs oranges I think. SD uses optimizations vs llama.cpp offloading more to CPU. Technically with CPP you can have no vram.

**Comment by muunkky** [comment]
> For one, image model vram usage is determined by the output size. While context will play a factor here, you can't really make an LLM take up less ram unless you use a smaller model or quantize to lower bits. Besides SD-XL, idle usage of SD is TINY.

I said ELI5... What do you mean by "While context will play a factor here, you can't really make an LLM take up less ram"

**Comment by Ph0rk0z** [comment]
The tricks from SD don't work on LLMs.

`--lowvram is a reimplementation of an optimization idea by [basujindal](https://github.com/basujindal/stable-diffusion). Model is separated into modules, and only one module is kept in GPU memory; when another module needs to run, the previous is removed from GPU memory. The nature of this optimization makes the processing run slower -- about 10 times slower compared to normal operation on my RTX 3090.`

That's basically cpu offloading or llama.cpp
